pkg/apis/psmdb/v1/psmdb_types.go

```diff
@@ -251,6 +251,23 @@ type PMMSpec struct {
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
 }
 
+func (spec *PMMSpec) IsEnabled(secret *corev1.Secret) bool {
```

Comment:
I would remove this method since I think it can be a bit confusing (it confused me while reviewing this).

I would expect that `IsEnabled()` is the same as `cr.Spec.PMM.Enabled` but it is not the case, `IsEnabled()` checks some additional thing, secrets in this case. In the future, this can potentially cause headaches, because someone can call this just to check if the thing is enabled (`Enabled` field) and it can be enabled but will the method can return false because it has no secrets.

Where you call this `IsEnabled()` simply check `cr.Spec.PMM.Enabled && spec.HasSecret(secret)`, it is not terrible :)

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1094/commits/31de53fa4ac86c40cddd2d27d0f88d674e054bd8